src/snowflake/snowpark/_internal/udf_utils.py

```diff
@@ -683,6 +683,11 @@ def create_python_udf_or_sp(
     strict: bool = False,
     secure: bool = False,
 ) -> None:
+    runtime_version = (
+        f"{sys.version_info[0]}.{sys.version_info[1]}"
+        if not session._runtime_version
+        else session._runtime_version
```

Comment:
Using an arbitrary runtime version only makes sense when `cloudpickle` is not involved (e.g. `register_from_file`). When `cloudpickle` is involved, we need to use the same python minor version between client/server such that SerDe works as expected. 

Should we add a check to make sure we are not giving user too much flexibility at the beginning? Being able to overwrite system runtime version is definitely useful ([SNOW-839835](https://snowflakecomputing.atlassian.net/browse/SNOW-839835)) :)

Reply:
sure, can force `_runtime_version` to be None, if `register_from_file` is called.

Reply:
A question here; when `cloudpickle` is involved, how is the Python version deduced

Reply:
Thx, updated the code to throw `ValueError` if the python runtime version mentioned via conda environment yaml file is different from the system runtime version, in the case where we are not registering from a file.